src/pubsub/ua_pubsub_manager.c

```diff
@@ -13,79 +13,107 @@
 
 #include "server/ua_server_internal.h"
 #include "ua_pubsub_ns0.h"
+#include "open62541/plugin/log_stdout.h"
 
```

Comment:
Don’t depend on specific plugins from the main library.

Reply:
You mean there should be <,> around this instead of " ?

Because this plugin is also used in e.g. "src/pubsub/ua_pubsub_config.c" for logging without the server context.

Reply:
Also we use all kinds of plugins in the main library, maybe I did not get the comment:

```shell
:open62541/src& ag --cc "#include <open62541/plugin"
ua_connection.c
21:#include <open62541/plugin/eventloop.h>

pubsub/ua_pubsub.h
15:#include <open62541/plugin/pubsub.h>

pubsub/ua_pubsub_networkmessage.h
14:#include <open62541/plugin/securitypolicy.h>

pubsub/ua_pubsub_config.c
9:#include <open62541/plugin/log_stdout.h>
13:#include <open62541/plugin/pubsub_ethernet.h>
15:#include <open62541/plugin/pubsub_udp.h>

pubsub/ua_pubsub_manager.c
16:#include <open62541/plugin/log_stdout.h>

ua_connection_internal.h
14:#include <open62541/plugin/network.h>
15:#include <open62541/plugin/eventloop.h>

server/ua_subscription.h
23:#include <open62541/plugin/nodestore.h>

server/ua_server_internal.h
22:#include <open62541/plugin/nodestore.h>

ua_securechannel.h
16:#include <open62541/plugin/log.h>
17:#include <open62541/plugin/securitypolicy.h>

server/ua_services_attribute.c
28:#include <open62541/plugin/historydatabase.h>
```

Reply:
Most of these are from /include/open62541/plugin.
These define the plug-in api itself.

a few of them are from /plugins/include/open62541/…
These are the plug-in implementations.
The plug-in implementations should be plugged into the config and used from there.
Hence the name…

Reply:
That clarifies it, so we should never be able to log something if there is no log context given. Only if it happens inside a plugin itself like e.g.

`plugins/ua_pubsub_ethernet.c`